test/unit_test_001.cpp

```diff
@@ -83,6 +86,50 @@ unittest(test_fletcher64)
 }
 
 
+unittest(test_FF_1)
+{
+  uint8_t x[1]  = { 0xFF };
+  uint16_t y[1] = { 0xFFFF };
+  uint32_t z[1] = { 0xFFFFFFFF };
+  
+  assertEqual(0, fletcher16(x, 1));
+  assertEqual(0, fletcher32(y, 1));
+  assertEqual(0, fletcher64(z, 1));
+}
+
+
+unittest(test_FF_2)
+{
+  uint8_t x[1]  = { 0xFF };
+  uint16_t y[1] = { 0xFFFF };
+  uint32_t z[1] = { 0xFFFFFFFF };
+
+  Fletcher16 FL16;
+  FL16.begin();
+  assertEqual(0, FL16.getFletcher());
+  FL16.begin(0xFF, 0);
```

Comment:
Just a comment: This `0xFF` has no effect, since `_s1    = (s1 == FLETCHER_16) ? 0 : s1;` reduces it to `0`.

Reply:
That means that it works as intended. If some (other) PR changes this I might get caught.
Unit tests must be written from the specification what a function/class should do, not because we know the implementation. If the specification of a function stays the same, one can optimize it etc. without changing the unit test.

(at least that is the theory)